README.md
# A2x1 Early Robotics

Provides Robots from beginning of the game. Reduces research requirements for Robotics and Logistcs technology

## Early Robotics Technology

Default value: Enabled

When enabled, All Robotics and Logistcs technologies have reduced research requirements and are researchable from beginning of the game

## Quick Start Player Inventory Items

Default value: Enabled

On beginning of the game places the following items in Player inventory

    small-electric-pole, count = 200
    offshore-pump, count = 1
    boiler, count = 4
    steam-engine, count = 8
    pipe, count = 10
    pipe-to-ground, count = 2
    coal, count = 100

    roboport, count = 10
    construction-robot, count = 100
    logistic-robot, count = 200
    logistic-chest-passive-provider, count = 50
    logistic-chest-requester, count = 50
    logistic-chest-storage, count = 5
    red-wire, count = 50

    assembling-machine-1, count = 20
    inserter, count = 200

    electric-mining-drill, count = 12
    stone-furnace, count = 4

## Roboport Multiplier Construction Radius

Default value: 10

Applies Multiplier on Roboport Construction Radius

## Roboport Multiplier Logistics Radius

Default value: 10

Applies Multiplier on Roboport Logistics Radius

## Roboport Multiplier Energy Usage

Default value: 0.1

Applies Multiplier on Robots Energy Usage Moving. Using 0.1 reduces its original value to 10%

## Robots Stack Size

Default value: 500

Updates All robots stack size

## Robots Multiplier Health

Default value: 10

Applies Multiplier on Robots Health

## Robots Multiplier Speed

Default value: 10

Applies Multiplier on Robots Speed

## Robots Multiplier Carry Size

Default value: 100

Applies Multiplier on Robots Carry Size

## Robots Multiplier Energy Usage Idle

Default value: 0.1

Applies Multiplier on Robots Energy Usage Idle. Using 0.1 reduces its original value to 10%

## Robots Multiplier Energy Usage Moving

Default value: 0.1

Applies Multiplier on Robots Energy Usage Moving. Using 0.1 reduces its original value to 10%


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.1.0
Date: 2021.02.11
---------------------------------------------------------------------------------------------------
Version: 1.1.0
Date: 2021.02.11
  Features:
    - Compatibility change for factorio 1.1
---------------------------------------------------------------------------------------------------
Version: 0.18.6
Date: 2020.07.07
  Changes:
    - Tweaked several default config values
---------------------------------------------------------------------------------------------------
Version: 0.18.5
Date: 2020.06.26
  Features:
    - Added Configurable Logistics Chest Extra Inventory Slots
---------------------------------------------------------------------------------------------------
Version: 0.18.4
Date: 2020.06.25
  Bugfixes:
    - You get Quick items only when setting is enabled
---------------------------------------------------------------------------------------------------
Version: 0.18.3
Date: 2020.06.21
  Bugfixes:
    - Corrected circuit-network technology ingredients
  Features:
    - Added red-wire as New Map start item
---------------------------------------------------------------------------------------------------
Version: 0.18.2
Date: 2020.06.21
  Info:
    - Added thumbnail.png
  Features:
    - Added logistic-chest-storage as New Map start item
---------------------------------------------------------------------------------------------------
Version: 0.18.1
Date: 2020.06.21
  Features:
    - Added configuration for Early Game Robotics and Logistcs technology research with reduced time and item requirements
    - Added configuration for Robots Health, Speed, Battery and Energy usage, Stack and Carry Size
    - Added configuration for Roboport Energy usage, Construction and Logistics Radius
    - Added ability to start new map with Robots in Player Inventory


settings.lua
data:extend(
  {
    {
      type = "string-setting",
      name = "a2x1_early_robotics-enable_early_robotics",
      setting_type = "startup",
      default_value = "enabled",
      allowed_values = {
        "enabled",
        "disabled"
      },
      order = 11
    },
    {
      type = "string-setting",
      name = "a2x1_early_robotics-enable_quick_start",
      setting_type = "startup",
      default_value = "enabled",
      allowed_values = {
        "enabled",
        "disabled"
      },
      order = 12
    },
    -- data_item_logistic_container
    {
      type = "double-setting",
      name = "a2x1_early_robotics-data_item_logistic_container-extra_inventory_slots",
      setting_type = "startup",
      default_value = 2,
      minimum_value = 0,
      order = 15
    },
    -- multipliers_roboport
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_roboport-construction_radius",
      setting_type = "startup",
      default_value = 5,
      order = 21
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_roboport-logistics_radius",
      setting_type = "startup",
      default_value = 5,
      order = 22
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_roboport-energy_usage",
      setting_type = "startup",
      default_value = 0.1,
      order = 23
    },
    -- multipliers_robot
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-stack_size",
      setting_type = "startup",
      default_value = 200,
      order = 31
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-max_health",
      setting_type = "startup",
      default_value = 10,
      order = 32
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-speed",
      setting_type = "startup",
      default_value = 10,
      order = 33
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-max_payload_size",
      setting_type = "startup",
      default_value = 25,
      order = 34
    },
    -- {
    --   type = "double-setting",
    --   name = "a2x1_early_robotics-multipliers_robot-max_energy",
    --   setting_type = "startup",
    --   default_value = 1,
    --   order = 35
    -- },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-energy_per_tick",
      setting_type = "startup",
      default_value = 0.1,
      order = 36
    },
    {
      type = "double-setting",
      name = "a2x1_early_robotics-multipliers_robot-energy_per_move",
      setting_type = "startup",
      default_value = 0.1,
      order = 37
    },
  }
)


resources/data_raw_recipe.lua
--
--

-- data.raw.recipe["flying-robot-frame"].enabled = false
-- data.raw.recipe["flying-robot-frame"].ingredients = {
--   {"electric-engine-unit", 1},
--   {"battery", 2},
--   {"steel-plate", 1},
--   {"electronic-circuit", 3}
-- }
-- data.raw.recipe["flying-robot-frame"].energy_required = 20
data.raw.recipe["flying-robot-frame"].energy_required = 10
data.raw.recipe["flying-robot-frame"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 10},
  {"iron-gear-wheel", 10},
  {"electronic-circuit", 3}
}

-- data.raw.recipe["roboport"].enabled = false
-- data.raw.recipe["roboport"].ingredients = {
--   {"steel-plate", 45},
--   {"iron-gear-wheel", 45},
--   {"advanced-circuit", 45}
-- }
data.raw.recipe["roboport"].ingredients = {
  {"iron-plate", 15},
  {"iron-gear-wheel", 15},
  {"electronic-circuit", 15}
}

-- data.raw.recipe["construction-robot"].enabled = false
-- data.raw.recipe["construction-robot"].ingredients = {
--   {"flying-robot-frame", 1},
--   {"electronic-circuit", 2}
-- }
data.raw.recipe["construction-robot"].ingredients = {
  {"flying-robot-frame", 1},
  {"electronic-circuit", 2}
}

-- data.raw.recipe["logistic-robot"].enabled = false
-- data.raw.recipe["logistic-robot"].ingredients = {
--   {"flying-robot-frame", 1},
--   {"advanced-circuit", 2}
-- }
data.raw.recipe["logistic-robot"].ingredients = {
  {"flying-robot-frame", 1},
  {"electronic-circuit", 2}
}

-- data.raw.recipe["logistic-chest-storage"].enabled = false
-- data.raw.recipe["logistic-chest-storage"].ingredients = {
--   {"steel-chest", 1},
--   {"electronic-circuit", 3},
--   {"advanced-circuit", 1}
-- }
data.raw.recipe["logistic-chest-storage"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 1},
  {"electronic-circuit", 3}
}

-- data.raw.recipe["logistic-chest-passive-provider"].enabled = false
-- data.raw.recipe["logistic-chest-passive-provider"].ingredients = {
--   {"steel-chest", 1},
--   {"electronic-circuit", 3},
--   {"advanced-circuit", 1}
-- }
data.raw.recipe["logistic-chest-passive-provider"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 1},
  {"electronic-circuit", 3}
}

-- data.raw.recipe["logistic-chest-active-provider"].enabled = false
-- data.raw.recipe["logistic-chest-active-provider"].ingredients = {
--   {"steel-chest", 1},
--   {"electronic-circuit", 3},
--   {"advanced-circuit", 1}
-- }
data.raw.recipe["logistic-chest-active-provider"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 1},
  {"electronic-circuit", 3}
}

-- data.raw.recipe["logistic-chest-requester"].enabled = false
-- data.raw.recipe["logistic-chest-requester"].ingredients = {
--   {"steel-chest", 1},
--   {"electronic-circuit", 3},
--   {"advanced-circuit", 1}
-- }
data.raw.recipe["logistic-chest-requester"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 1},
  {"electronic-circuit", 3}
}

-- data.raw.recipe["logistic-chest-buffer"].enabled = false
-- data.raw.recipe["logistic-chest-buffer"].ingredients = {
--   {"steel-chest", 1},
--   {"electronic-circuit", 3},
--   {"advanced-circuit", 1}
-- }
data.raw.recipe["logistic-chest-buffer"].ingredients = {
  {"iron-chest", 1},
  {"iron-plate", 1},
  {"electronic-circuit", 3}
}


resources/data_technology.lua
--
--
-- YAY works
--
--

-- Detach "personal-roboport-equipment"
data.raw.technology["personal-roboport-equipment"].prerequisites = {"solar-panel-equipment"}

-- Detach "utility-science-pack" and "worker-robots-#3" upgrades
data.raw.technology["utility-science-pack"].prerequisites = {"battery", "electric-engine"}
data.raw.technology["worker-robots-speed-3"].prerequisites = {"utility-science-pack"}
data.raw.technology["worker-robots-storage-3"].prerequisites = {"utility-science-pack"}

-- Move "circuit-network" under "automation"
data.raw.technology["circuit-network"].prerequisites = {"automation"}
data.raw.technology["circuit-network"].unit.time = 10 --15
data.raw.technology["circuit-network"].unit.count = 10 --100
data.raw.technology["circuit-network"].ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1},
}

-- Move "robotics" under "automation" and "logistics"
data.raw.technology["robotics"].prerequisites = {"automation", "logistics"}
data.raw.technology["robotics"].unit.time = 10 --30
data.raw.technology["robotics"].unit.count = 10 --75
data.raw.technology["robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1},
  -- {"chemical-science-pack", 1}
}

data.raw.technology["construction-robotics"].prerequisites = {"robotics"}
data.raw.technology["construction-robotics"].unit.time = 10 --30
data.raw.technology["construction-robotics"].unit.count = 10 --100
data.raw.technology["construction-robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
}

data.raw.technology["logistic-robotics"].prerequisites = {"robotics"}
data.raw.technology["logistic-robotics"].unit.time = 10 --30
data.raw.technology["logistic-robotics"].unit.count = 10 --250
data.raw.technology["logistic-robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
}

data.raw.technology["logistic-system"].prerequisites = {"logistic-robotics"}
data.raw.technology["logistic-system"].unit.time = 10 --30
data.raw.technology["logistic-system"].unit.count = 10 --500
data.raw.technology["logistic-system"].unit.ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
  -- {"utility-science-pack", 1}
}

data.raw.technology["worker-robots-storage-1"].prerequisites = {"robotics"}
-- data.raw.technology["worker-robots-storage-1"].unit.time = 30
-- data.raw.technology["worker-robots-storage-1"].unit.count = 200
data.raw.technology["worker-robots-storage-1"].ingredients = {
  {"automation-science-pack", 1}
  -- {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
}

data.raw.technology["worker-robots-storage-2"].prerequisites = {"logistic-science-pack", "worker-robots-storage-1"}
-- data.raw.technology["worker-robots-storage-2"].unit.time = 60
-- data.raw.technology["worker-robots-storage-2"].unit.count = 300
data.raw.technology["worker-robots-storage-2"].ingredients = {
  {"automation-science-pack", 1},
  {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
  -- {"production-science-pack", 1}
}

data.raw.technology["worker-robots-speed-1"].prerequisites = {"robotics"}
-- data.raw.technology["worker-robots-speed-1"].unit.time = 30
-- data.raw.technology["worker-robots-speed-1"].unit.count = 50
data.raw.technology["worker-robots-speed-1"].ingredients = {
  {"automation-science-pack", 1},
  {"logistic-science-pack", 1}
  -- {"chemical-science-pack", 1}
}


resources/data_raw_updates.lua
--
--
-- https://lua-api.factorio.com/latest/LuaEntityPrototype.html
--
--
-- speed            - The default speed of this flying robot, rolling stock, unit or nil.
-- max_health       - Max health of this entity.
-- max_payload_size - The max payload size of this logistics or construction robot or nil.
-- max_energy       - The max energy for this flying robot or nil.
-- energy_usage     - The direct energy usage of this entity or nil if this entity doesn't have a direct energy usage.
-- energy_per_move  - The energy consumed per tile moved for this flying robot or nil.
-- energy_per_tick  - The energy consumed per tick for this flying robot or nil.

for i, v in pairs(data.raw["construction-robot"]) do
  v.speed = v.speed * settings.startup["a2x1_early_robotics-multipliers_robot-speed"].value
  v.max_health = v.max_health * settings.startup["a2x1_early_robotics-multipliers_robot-max_health"].value
  v.max_payload_size = v.max_payload_size * settings.startup["a2x1_early_robotics-multipliers_robot-max_payload_size"].value

  -- local max_energy_num, max_energy_unit = v.max_energy:match("(%d+%.?%d*)(%a+)")
  -- v.max_energy = max_energy_num * settings.startup["a2x1_early_robotics-multipliers_robot-max_energy"].value .. max_energy_unit

  local energy_per_tick_num, energy_per_tick_unit = v.energy_per_tick:match("(%d+%.?%d*)(%a+)")
  v.energy_per_tick = energy_per_tick_num * settings.startup["a2x1_early_robotics-multipliers_robot-energy_per_tick"].value .. energy_per_tick_unit

  local energy_per_move_num, energy_per_move_unit = v.energy_per_move:match("(%d+%.?%d*)(%a+)")
  v.energy_per_move = energy_per_move_num * settings.startup["a2x1_early_robotics-multipliers_robot-energy_per_move"].value .. energy_per_move_unit

  data.raw["item"][v.name].stack_size = settings.startup["a2x1_early_robotics-multipliers_robot-stack_size"].value
end

for i, v in pairs(data.raw["logistic-robot"]) do
  v.speed = v.speed * settings.startup["a2x1_early_robotics-multipliers_robot-speed"].value
  v.max_health = v.max_health * settings.startup["a2x1_early_robotics-multipliers_robot-max_health"].value
  v.max_payload_size = v.max_payload_size * settings.startup["a2x1_early_robotics-multipliers_robot-max_payload_size"].value

  -- local max_energy_num, max_energy_unit = v.max_energy:match("(%d+%.?%d*)(%a+)")
  -- v.max_energy = max_energy_num * settings.startup["a2x1_early_robotics-multipliers_robot-max_energy"].value .. max_energy_unit

  local energy_per_tick_num, energy_per_tick_unit = v.energy_per_tick:match("(%d+%.?%d*)(%a+)")
  v.energy_per_tick = energy_per_tick_num * settings.startup["a2x1_early_robotics-multipliers_robot-energy_per_tick"].value .. energy_per_tick_unit

  local energy_per_move_num, energy_per_move_unit = v.energy_per_move:match("(%d+%.?%d*)(%a+)")
  v.energy_per_move = energy_per_move_num * settings.startup["a2x1_early_robotics-multipliers_robot-energy_per_move"].value .. energy_per_move_unit

  data.raw["item"][v.name].stack_size = settings.startup["a2x1_early_robotics-multipliers_robot-stack_size"].value
end

for i, v in pairs(data.raw["roboport"]) do
  v.construction_radius = v.construction_radius * settings.startup["a2x1_early_robotics-multipliers_roboport-construction_radius"].value
  v.logistics_radius = v.logistics_radius * settings.startup["a2x1_early_robotics-multipliers_roboport-logistics_radius"].value

  local energy_usage_num, energy_usage_unit = v.energy_usage:match("(%d+%.?%d*)(%a+)")
  v.energy_usage = energy_usage_num * settings.startup["a2x1_early_robotics-multipliers_roboport-energy_usage"].value .. energy_usage_unit
end


resources/data_raw_technology.lua
--
--

-- Detach "personal-roboport-equipment"
data.raw.technology["personal-roboport-equipment"].prerequisites = {"solar-panel-equipment"}

-- Detach "utility-science-pack" and "worker-robots-#3" upgrades
data.raw.technology["utility-science-pack"].prerequisites = {"battery", "electric-engine"}
data.raw.technology["worker-robots-speed-3"].prerequisites = {"utility-science-pack"}
data.raw.technology["worker-robots-storage-3"].prerequisites = {"utility-science-pack"}

-- Move "circuit-network" under "automation"
data.raw.technology["circuit-network"].prerequisites = {"automation"}
-- data.raw.technology["circuit-network"].unit.time = 15
data.raw.technology["circuit-network"].unit.time = 10
-- data.raw.technology["circuit-network"].unit.count = 100
data.raw.technology["circuit-network"].unit.count = 10
-- data.raw.technology["circuit-network"].ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1}
-- }
data.raw.technology["circuit-network"].unit.ingredients = {
  {"automation-science-pack", 1}
}

-- Move "robotics" under "automation" and "logistics"
data.raw.technology["robotics"].prerequisites = {"automation", "logistics"}
-- data.raw.technology["robotics"].unit.time = 30
data.raw.technology["robotics"].unit.time = 10
-- data.raw.technology["robotics"].unit.count = 75
data.raw.technology["robotics"].unit.count = 10
-- data.raw.technology["robotics"].unit.ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1}
-- }
data.raw.technology["robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
}

data.raw.technology["construction-robotics"].prerequisites = {"robotics"}
-- data.raw.technology["construction-robotics"].unit.time = 30
data.raw.technology["construction-robotics"].unit.time = 10
-- data.raw.technology["construction-robotics"].unit.count = 100
data.raw.technology["construction-robotics"].unit.count = 10
-- data.raw.technology["construction-robotics"].unit.ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1}
-- }
data.raw.technology["construction-robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
}

data.raw.technology["logistic-robotics"].prerequisites = {"robotics"}
-- data.raw.technology["logistic-robotics"].unit.time = 30
data.raw.technology["logistic-robotics"].unit.time = 10
-- data.raw.technology["logistic-robotics"].unit.count = 250
data.raw.technology["logistic-robotics"].unit.count = 10
-- data.raw.technology["logistic-robotics"].unit.ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1}
-- }
data.raw.technology["logistic-robotics"].unit.ingredients = {
  {"automation-science-pack", 1}
}

data.raw.technology["logistic-system"].prerequisites = {"logistic-robotics"}
-- data.raw.technology["logistic-system"].unit.time = 30
data.raw.technology["logistic-system"].unit.time = 10
-- data.raw.technology["logistic-system"].unit.count = 500
data.raw.technology["logistic-system"].unit.count = 10
-- data.raw.technology["logistic-system"].unit.ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1},
--   {"utility-science-pack", 1}
-- }
data.raw.technology["logistic-system"].unit.ingredients = {
  {"automation-science-pack", 1}
}

data.raw.technology["worker-robots-storage-1"].prerequisites = {"robotics"}
-- data.raw.technology["worker-robots-storage-1"].unit.time = 30
-- data.raw.technology["worker-robots-storage-1"].unit.count = 200
-- data.raw.technology["worker-robots-storage-1"].ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1}
-- }
data.raw.technology["worker-robots-storage-1"].ingredients = {
  {"automation-science-pack", 1}
}

data.raw.technology["worker-robots-storage-2"].prerequisites = {"logistic-science-pack", "worker-robots-storage-1"}
-- data.raw.technology["worker-robots-storage-2"].unit.time = 60
-- data.raw.technology["worker-robots-storage-2"].unit.count = 300
-- data.raw.technology["worker-robots-storage-2"].ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1},
--   {"production-science-pack", 1}
-- }
data.raw.technology["worker-robots-storage-2"].ingredients = {
  {"automation-science-pack", 1},
  {"logistic-science-pack", 1}
}

data.raw.technology["worker-robots-speed-1"].prerequisites = {"robotics"}
-- data.raw.technology["worker-robots-speed-1"].unit.time = 30
-- data.raw.technology["worker-robots-speed-1"].unit.count = 50
-- data.raw.technology["worker-robots-speed-1"].ingredients = {
--   {"automation-science-pack", 1},
--   {"logistic-science-pack", 1},
--   {"chemical-science-pack", 1}
-- }
data.raw.technology["worker-robots-speed-1"].ingredients = {
  {"automation-science-pack", 1},
  {"logistic-science-pack", 1}
}


control.lua
script.on_event(
  defines.events.on_player_created,
  function(event)
    if settings.startup["a2x1_early_robotics-enable_quick_start"].value == "enabled"  then
      local player = game.players[event.player_index]
      player.insert {name = "small-electric-pole", count = 200}
      player.insert {name = "offshore-pump", count = 1}
      player.insert {name = "boiler", count = 4}
      player.insert {name = "steam-engine", count = 8}
      player.insert {name = "pipe", count = 10}
      player.insert {name = "pipe-to-ground", count = 2}
      player.insert {name = "coal", count = 100}

      player.insert {name = "roboport", count = 10}
      player.insert {name = "construction-robot", count = 100}
      player.insert {name = "logistic-robot", count = 200}
      player.insert {name = "logistic-chest-passive-provider", count = 50}
      player.insert {name = "logistic-chest-requester", count = 50}
      player.insert {name = "logistic-chest-storage", count = 5}
      player.insert {name = "red-wire", count = 50}

      player.insert {name = "assembling-machine-1", count = 20}
      player.insert {name = "inserter", count = 200}

      player.insert {name = "electric-mining-drill", count = 12}
      player.insert {name = "stone-furnace", count = 4}
    end
  end
)
